test/natalie/array_test.rb

```diff
@@ -44,6 +44,20 @@
     end
   end
 
+  describe 'hash' do
+    it 'does not collide when array contains duplicate elements' do
+      found_hashes = []
+      (0..100).map do |length| 
+        current_hash = (0..length).to_a.map { |x| 0 }.hash 
+        if found_hashes.find_index current_hash
+          fail
+        else
+          found_hashes << current_hash
+        end 
+      end
+    end
+  end
```

Comment:
Can you move this down with the other `describe '#hash'` block?

Reply:
Oooops sorry, will do!

Reply:
fixed, thanks ;)